src/Feature.tsx

```diff
@@ -14,157 +14,37 @@
+    return null;
   }
 
-  componentDidMount() {
-    const {
-      feature,
-      optimizely,
-      optimizelyReadyTimeout,
-      isServerSide,
-      timeout,
-    } = this.props
-    if (!optimizely) {
-      throw new Error('optimizely prop must be supplied')
-    }
-
-    if (isServerSide) {
-      return
-    }
-
-    // allow overriding of the ready timeout via the `timeout` prop passed to <Experiment />
-    let finalReadyTimeout: number | undefined =
-      timeout !== undefined ? timeout : optimizelyReadyTimeout
-
-    optimizely.onReady({ timeout: finalReadyTimeout }).then((res: OnReadyResult) => {
-      if (res.success) {
-        logger.info('feature="%s" successfully rendered for user="%s"', feature, optimizely.user.id)
-      } else {
-        logger.info(
-          'feature="%s" could not be checked before timeout of %sms, reason="%s" ',
-          feature,
-          timeout === undefined ? DEFAULT_ON_READY_TIMEOUT : timeout,
-          res.reason || '',
-        )
-      }
-
-      const isEnabled = optimizely.isFeatureEnabled(feature)
-      const variables = optimizely.getFeatureVariables(feature)
-      this.setState({
-        canRender: true,
-        isEnabled,
-        variables,
-      })
-
-      if (this.autoUpdate) {
-        this.setupAutoUpdateListeners()
-      }
-    })
-  }
-
-  setupAutoUpdateListeners() {
-    const { optimizely, feature } = this.props
-    if (optimizely === null) {
-      return
-    }
-
-    this.optimizelyNotificationId = optimizely.notificationCenter.addNotificationListener(
-      'OPTIMIZELY_CONFIG_UPDATE',
-      () => {
-        logger.info('OPTIMIZELY_CONFIG_UPDATE, re-evaluating feature="%s" for user="%s"', feature, optimizely.user.id)
-        const isEnabled = optimizely.isFeatureEnabled(feature)
-        const variables = optimizely.getFeatureVariables(feature)
-        this.setState({
-          isEnabled,
-          variables,
-        })
-      },
-    )
-
-    this.unregisterUserListener = optimizely.onUserUpdate(() => {
-      logger.info('User update, re-evaluating feature="%s" for user="%s"', feature, optimizely.user.id)
-      const isEnabled = optimizely.isFeatureEnabled(feature)
-      const variables = optimizely.getFeatureVariables(feature)
-      this.setState({
-        isEnabled,
-        variables,
-      })
-    })
-  }
-
-  componentWillUnmount() {
-    const { optimizely, isServerSide } = this.props
-    if (isServerSide || !this.autoUpdate) {
-      return
-    }
-    if (optimizely && this.optimizelyNotificationId) {
-      optimizely.notificationCenter.removeNotificationListener(
-        this.optimizelyNotificationId,
-      )
-    }
-    this.unregisterUserListener()
-  }
-
-  render() {
-    const { children } = this.props
-    const { isEnabled, variables, canRender } = this.state
-
-    if (!canRender) {
-      return null
-    }
-
-    return children(isEnabled, variables)
-  }
-}
+  return children(isEnabled, variables)
```

Comment:
Is there value in passing `clientReady` and `didTimeout` like `useFeature` has? One usecase could be the decision to show a spinner while disabled until the client is ready then to show the default case.

Reply:
I like this idea. Will add it in